ts/src/pro/bitmex.ts

```diff
@@ -603,6 +604,212 @@ export default class bitmex extends bitmexRest {
         }
     }
 
+    async watchPositions (symbols: Strings = undefined, since: Int = undefined, limit: Int = undefined, params = {}) {
```

Comment:
@pcriadoperez it crashed when I was testing it out

```
side': 'long',
  'stopLossPrice': None,
  'symbol': 'BTC/USDT:USDT',
  'takeProfitPrice': None,
  'timestamp': 1702022400208,
  'unrealizedPnl': '0'}]
Traceback (most recent call last):
  File "/Users/cjg/Git/ccxt10/ccxt/examples/py/cli.py", line 241, in <module>
    asyncio.run(main())
  File "/opt/homebrew/Cellar/python@3.11/3.11.5/Frameworks/Python.framework/Versions/3.11/lib/python3.11/asyncio/runners.py", line 190, in run
    return runner.run(main)
           ^^^^^^^^^^^^^^^^
  File "/opt/homebrew/Cellar/python@3.11/3.11.5/Frameworks/Python.framework/Versions/3.11/lib/python3.11/asyncio/runners.py", line 118, in run
    return self._loop.run_until_complete(task)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/Cellar/python@3.11/3.11.5/Frameworks/Python.framework/Versions/3.11/lib/python3.11/asyncio/base_events.py", line 653, in run_until_complete
    return future.result()
           ^^^^^^^^^^^^^^^
  File "/Users/cjg/Git/ccxt10/ccxt/examples/py/cli.py", line 220, in main
    result = await result
             ^^^^^^^^^^^^
  File "/Users/cjg/Git/ccxt10/ccxt/python/ccxt/pro/bitmex.py", line 610, in watch_positions
    newPositions = await self.watch(url, messageHash, request, subscriptionHash)
                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/cjg/Git/ccxt10/ccxt/python/ccxt/async_support/base/ws/fast_client.py", line 27, in handler
    self.handle_message(message)
  File "/Users/cjg/Git/ccxt10/ccxt/python/ccxt/async_support/base/ws/aiohttp_client.py", line 34, in handle_message
    self.handle_text_or_binary_message(message.data)
  File "/Users/cjg/Git/ccxt10/ccxt/python/ccxt/async_support/base/ws/aiohttp_client.py", line 29, in handle_text_or_binary_message
    self.on_message_callback(self, decoded)
  File "/Users/cjg/Git/ccxt10/ccxt/python/ccxt/pro/bitmex.py", line 1529, in handle_message
    return method(client, message)
           ^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/cjg/Git/ccxt10/ccxt/python/ccxt/pro/bitmex.py", line 771, in handle_positions
    position = self.parse_position(rawPosition)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/cjg/Git/ccxt10/ccxt/python/ccxt/async_support/bitmex.py", line 2121, in parse_position
    notionalString = Precise.string_abs(self.safe_string(position, 'foreignNotional', 'homeNotional'))
                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/cjg/Git/ccxt10/ccxt/python/ccxt/base/precise.py", line 227, in string_abs
    return str(Precise(string).abs())
               ^^^^^^^^^^^^^^^
  File "/Users/cjg/Git/ccxt10/ccxt/python/ccxt/base/precise.py", line 21, in __init__
    modifier = int(modifier)
               ^^^^^^^^^^^^^
```

Reply:
hmm strange I stried in python both BTC/USD:BTC and BTC/USDT:USDT and both worked for me:
```py
[{'collateral': None,
  'contractSize': 1e-06,
  'contracts': 1000.0,
  'datetime': '2023-12-11T20:23:04.740Z',
  'entryPrice': None,
  'hedged': None,
  'id': '412475',
  'info': {'account': 412475,
           'currency': 'USDt',
           'currentQty': 1000,
           'foreignNotional': -40.91912,
           'liquidationPrice': 0.0,
           'maintMargin': 546650,
           'markPrice': 40919.12,
           'markValue': 40919120,
           'riskValue': 40919120,
           'symbol': 'XBTUSDT',
           'timestamp': '2023-12-11T20:23:04.740Z',
           'unrealisedPnl': 107620,
           'unrealisedRoePcnt': 0.2637},
  'initialMargin': None,
  'initialMarginPercentage': None,
  'lastPrice': None,
  'lastUpdateTimestamp': None,
  'leverage': None,
  'liquidationPrice': 0.0,
  'maintenanceMargin': '0.54665',
  'maintenanceMarginPercentage': None,
  'marginMode': 'isolated',
  'marginRatio': None,
  'markPrice': 40919.12,
  'notional': 40.91912,
  'percentage': None,
  'side': None,
  'stopLossPrice': None,
  'symbol': 'BTC/USDT:USDT',
  'takeProfitPrice': None,
  'timestamp': 1702326184740,
  'unrealizedPnl': '0.10762'}]
  ```

Reply:
I wasn't able to reproduce but from the error log you posted I think I can see where it happens. I'll post a fix.

Reply:
@pcriadoperez No, I was using the sandbox as well

Reply:
@pcriadoperez this is easy to reproduce:
- start watching Positions ( p bitmex watchPositions --sandbox)
- change the position value (bitmex createOrder "SOL/USDT:USDT" market buy 1000  --sandbox)

Reply:
hum now that I'm debugging it, it does not crash...

```
➜  ccxt git:(bitmex/watchPositions) ✗  cd /Users/cjg/Git/ccxt10/ccxt ; /usr/bin/env /opt/homebrew/bin/python3 /Users/cjg/.vscode/extensions/ms-python.p
ython-2023.22.0/pythonFiles/lib/python/debugpy/adapter/../../debugpy/launcher 63668 -- /Users/cjg/Git/ccxt10/ccxt/examples/py/bybit-updated.py 
CCXT Version: 4.1.84
notional::-7.0771
notional::0.0
notional::0.0
notional::-7.0789
notional::-7.0776
notional::-14.1552
notional::-14.152
notional::-7.076
notional::-7.0759
notional::-7.0758
notional::-7.0752
notional::-7.0751
notional::-14.1502
notional::-14.1468
notional::-14.14
notional::-7.07
notional::-7.0689
```

Reply:
@pcriadoperez I found a different issue, after some time running the connection closes with a 1006 error, are we missing a ping or smth?

Reply:
for some reason isn't happening anymore..

Reply:
@carlosmiei , yes tried testing it from my side and stay subscribed for 30 minutes and didn't hit any 1006 error